src/main/scala/courses/domain/repositories/UsersRepository.scala

```diff
@@ -0,0 +1,30 @@
+package courses.domain.repositories
+
+import courses.config.DatabaseConfig
+import courses.domain.models.User
+import courses.domain.tables.UserTable
+import courses.utils.Retrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait UserRepository[F[_]] {
+  def register(username: String, address: Option[String], email: String): F[User]
+  def getById(id: Long): F[Option[User]]
+  def getByUsername(username: String): F[Option[User]]
+}
+
+class UsersRepository(implicit ec: ExecutionContext) extends UserRepository[Future] with UserTable with DatabaseConfig {
+
+  import scala.concurrent.duration._
+
+  def register(username: String, address: Option[String], email: String): Future[User] = {
+    val action = insertAction += User(0, username, address, email)
+    // can't use retrier, because of type mismatch
+//    Retrier.retryAsync[User](db.run(action), res => res != null, List(1.second))
```

Comment:
what type? you should use slightly modified version of retry async which retries some DB SQL exceptions.